tests/e2e/e2e_permissive_smi_switching.go

```diff
@@ -0,0 +1,290 @@
+				svcAccDef, deploymentDef, svcDef := Td.SimpleDeploymentApp(
+					SimpleDeploymentAppDef{
+						Name:         "server",
+						Namespace:    destApp,
+						ReplicaCount: int32(replicaSetPerService),
+						Image:        "kennethreitz/httpbin",
+						Ports:        []int{DefaultUpstreamServicePort},
+						Command:      HttpbinCmd,
+					})
+
+				_, err := Td.CreateServiceAccount(destApp, &svcAccDef)
+				Expect(err).NotTo(HaveOccurred())
+				_, err = Td.CreateDeployment(destApp, deploymentDef)
+				Expect(err).NotTo(HaveOccurred())
+				_, err = Td.CreateService(destApp, svcDef)
+				Expect(err).NotTo(HaveOccurred())
+
+				wg.Add(1)
+				go func(wg *sync.WaitGroup, srcClient string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					Expect(Td.WaitForPodsRunningReady(destApp, 200*time.Second, replicaSetPerService, nil)).To(Succeed())
+				}(&wg, destApp)
+
+				// Create all client deployments, also with replicaset
+				for _, srcClient := range sourceNamespaces {
+					svcAccDef, deploymentDef, svcDef = Td.SimpleDeploymentApp(
+						SimpleDeploymentAppDef{
+							Name:         srcClient,
+							Namespace:    srcClient,
+							ReplicaCount: int32(replicaSetPerService),
+							Command:      []string{"/bin/bash", "-c", "--"},
+							Args:         []string{"while true; do sleep 30; done;"},
+							Image:        "songrgg/alpine-debug",
+							Ports:        []int{DefaultUpstreamServicePort}, // Can't deploy services with empty/no ports
+						})
+					_, err = Td.CreateServiceAccount(srcClient, &svcAccDef)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = Td.CreateDeployment(srcClient, deploymentDef)
+					Expect(err).NotTo(HaveOccurred())
+					_, err = Td.CreateService(srcClient, svcDef)
+					Expect(err).NotTo(HaveOccurred())
+
+					wg.Add(1)
+					go func(wg *sync.WaitGroup, srcClient string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						Expect(Td.WaitForPodsRunningReady(srcClient, 200*time.Second, replicaSetPerService, nil)).To(Succeed())
+					}(&wg, srcClient)
+				}
+				wg.Wait()
+
+				// Create Multiple HTTP request structure and fill it with appropriate details
+				requests := HTTPMultipleRequest{
+					Sources: []HTTPRequestDef{},
+				}
+				for _, ns := range sourceNamespaces {
+					pods, err := Td.Client.CoreV1().Pods(ns).List(context.Background(), metav1.ListOptions{})
+					Expect(err).To(BeNil())
+
+					for _, pod := range pods.Items {
+						requests.Sources = append(requests.Sources, HTTPRequestDef{
+							SourceNs:        ns,
+							SourcePod:       pod.Name,
+							SourceContainer: ns, // container_name == NS for this test
+
+							Destination: fmt.Sprintf("%s.%s:%d", destApp, destApp, DefaultUpstreamServicePort),
+						})
+					}
+				}
+
+				var results HTTPMultipleResults
+
+				By("Test and expect traffic failure with no permissive mode and no SMI rules")
+				success := Td.WaitForRepeatedSuccess(func() bool {
```

Comment:
For consistency, instead of switching between SMI and permissive mode and repeating the code segments that tests the HTTP requests, we could keep the requests segment as it's own function /ginkgo context and pass in a parameter for permissive mode like we do in the e2e_tcp_client_server test: https://github.com/openservicemesh/osm/blob/main/tests/e2e/e2e_tcp_client_server_test.go#L24-L29

Reply:
you can't use contexts, there's cleanup between them.
I could do it between `It()`s?

Reply:
I can probably move the success/failure traffic checks on a function though, I was lazy

Reply:
@nojnhuh is there cleanup between `It()`s?

Reply:
I'm almost positive there is, but I would have to double check to be sure.